### ISSUE TITLE
Render Charts to SVG server side with d3 node module.

### ISSUE THREAD

**Issue by bwklein** [asciidoctor/asciidoctor-chart#7]
Please see this page for a working example.
[Render SVG Charts Server-side With D3.js](http://www.gregjopa.com/2013/11/render-svg-charts-server-side-with-d3-js/)

If this could be used to take a chart to svg file and then insert it into a document, it would be a great solution for both the HTML output and to PDF files.

**Comment by Mogztter** [comment]
Related issue: https://github.com/Mogztter/asciidoctor-chart/issues/1

The main challenge is that most charting libraries rely on JavaScript/DOM.
Is there any pure Ruby charting library? I found  https://github.com/topfunky/gruff but it relies on ImageMagick...

**Comment by Mogztter** [comment]
Would it be acceptable to use https://github.com/rubycdp/ferrum to interact with a Chrome headless instance and generate SVG? When transpiling to JavaScript we could implement the same logic using Puppeteer (i.e., replace Ferrum by Puppeteer).

Since Ferrum claims to be a pure Ruby library, it might also works seamlessly on the JVM with JRuby.

**Comment by cgoguyer** [comment]
If there any news on rendering chart inside PDF document ?

**Comment by Mogztter** [comment]
Nope, I haven't found a good solution to implement this feature... using a Chrome instance seems a bit heavyweight! 
If you have a better idea, feel free to suggest it or even better submit a pull request.

As an alternative, you can try https://github.com/mogztter/asciidoctor-web-pdf which relies on web technologies to produce a PDF.
